src/arrays.jl

```diff
@@ -320,6 +320,46 @@ end
 Base.fill(x::UnionAbstractQuantity, dims::Dims...) = QuantityArray(fill(ustrip(x), dims...), dimension(x), typeof(x))
 Base.fill(x::UnionAbstractQuantity, t::Tuple{}) = QuantityArray(fill(ustrip(x), t), dimension(x), typeof(x))
 
+# Will be overloaded by `DynamicQuantitiesLinearAlgebraExt`:
+_norm(_) = error("Please load the `LinearAlgebra.jl` package.")
+
+# Define isapprox for vectors of Quantity's
+function Base.isapprox(
+    u::AbstractArray{Q1},
+    v::AbstractArray{Q2};
+    atol=new_quantity(Q1, zero(T1), dimension(first(u))),
+    rtol=Base.rtoldefault(T1),
+    nans::Bool=false,
+    norm::F=_norm
+) where {T1,D1,Q1<:UnionAbstractQuantity{T1,D1},T2,D2,Q2<:UnionAbstractQuantity{T2,D2},F<:Function}
+    d = norm(u - v)
```

Comment:
Presumably we also want to fall back to component-wise comparison when dimensions are not equal across the array?

Reply:
Pushed code for this.

Reply:
The way that I laid out the specialized `isapprox` methods in this branch was intended to mirror how things are defined upstream.

- Julia Base only defines `isapprox` as a 1:1 scalar comparison and doesn't attempt to handle vector comparisons, but users are free to perform component-wise comparisons via broadcasting.
- LinearAlgebra.jl introduces a method for vector comparison, where the delta isn't component-wise, rather it assesses similarity based on a threshold for the `norm` of the "tip-to-tip" vector difference. This works well for vectors composed of simple scalar values and vector spaces with homogeneous units/dimensions, but isn't well-defined for vectors containing heterogeneous units/dimensions (what's the norm?).

Based on these upstream patterns and this mixed-units issue, I feel like not defining a component-wise fallback is probably the "correct" solution. That being said, the pragmatic side of me recognizes that users probably aren't thinking this deeply about what it means for two vectors to be approximately equal, and so would be confused by a `DimensionError` when trying to compute an `isapprox` without broadcasting.

Reply:
> Based on these upstream patterns and this mixed-units issue, I feel like not defining a component-wise fallback is probably the "correct" solution.

I agree with this. Users can always do `.≈` if needed. We can also just tag the `DimensionError` with some extra debugging info for users. Generally I do think its better to error loudly and obviously (with good error messages).